openwebstart/src/main/java/com/openwebstart/app/icon/ApplicationIconCache.java

```diff
@@ -63,7 +57,7 @@ public synchronized void triggerDownload(final Application application, final Ic
             } else {
                 final CompletableFuture<BufferedImage> result = new CompletableFuture<>();
                 inProgressMap.put(iconDescription, result);
-                executor.execute(() -> {
+                getDaemonExecutorService().execute(() -> {
```

Comment:
It won't be a problem to have this one as a non daemon executor. I prefer to change as less behaviour as possible with this PR so we should change it here to non daemon.